src/Microsoft.AspNet.Http.Features/FeatureReferences.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNet.Http.Features
+{
+    public struct FeatureReferences<TCache>
+    {
+        public FeatureReferences(IFeatureCollection collection)
+        {
+            Collection = collection;
+            Cache = default(TCache);
+            Revision = collection.Revision;
+        }
+
+        public readonly IFeatureCollection Collection;
```

Comment:
If these are fields instead of properties for a specific reason you should put that in a comment.